src/main/java/org/jsmart/zerocode/core/engine/mocker/RestEndPointMocker.java

```diff
@@ -1,51 +1,119 @@
+
+        restartWireMock(mockPort);
 
-        // TODO - Check wiremock APIs to whether it supports method as a paramater or direct JSON
         mockSteps.getMocks().forEach(mockStep -> {
             String jsonBodyRequest = mockStep.getResponse().get("body").toString();
 
             if("GET".equals(mockStep.getOperation())){
-                givenThat(WireMock.get(urlEqualTo(mockStep.getUrl()))
-                        .willReturn(aResponse()
-                                .withStatus(mockStep.getResponse().get("status").asInt())
-                                .withHeader("Content-Type", APPLICATION_JSON)
-                                .withBody(jsonBodyRequest)));
+                LOGGER.info("*****WireMock- Mocking the GET endpoint");
+                givenThat(createGetRequestBuilder(mockStep)
+                        .willReturn(responseBuilder(mockStep, jsonBodyRequest)));
+                LOGGER.info("WireMock- Mocking the GET endpoint -done- *****");
             }
             else if("POST".equals(mockStep.getOperation())){
-                givenThat(WireMock.post(urlEqualTo(mockStep.getUrl()))
-                        .willReturn(aResponse()
-                                .withStatus(mockStep.getResponse().get("status").asInt())
-                                .withHeader("Content-Type", APPLICATION_JSON)
-                                .withBody(jsonBodyRequest)));
+                LOGGER.info("*****WireMock- Mocking the POST endpoint");
+                givenThat(createPostRequestBuilder(mockStep)
+                        .willReturn(responseBuilder(mockStep, jsonBodyRequest)));
+                LOGGER.info("WireMock- Mocking the POST endpoint -done-*****");
             }
             else if("PUT".equals(mockStep.getOperation())){
-                givenThat(WireMock.put(urlEqualTo(mockStep.getUrl()))
-                        .willReturn(aResponse()
-                                .withStatus(mockStep.getResponse().get("status").asInt())
-                                .withHeader("Content-Type", APPLICATION_JSON)
-                                .withBody(jsonBodyRequest)));
+                LOGGER.info("*****WireMock- Mocking the PUT endpoint");
+                givenThat(createPutRequestBuilder(mockStep)
+                        .willReturn(responseBuilder(mockStep, jsonBodyRequest)));
+                LOGGER.info("WireMock- Mocking the PUT endpoint -done-*****");
             }
         });
     }
 
+
+    public static void restartWireMock(int dynamicPort) {
+        wireMockServer = new WireMockServer(wireMockConfig().port(dynamicPort)); // <-- Strange
+        /*
+         * Stop the wireMock server if it is running previously due to any other tests.
+         */
+        wireMockServer.stop();
+        wireMockServer.start();
+        WireMock.configureFor("localhost", dynamicPort); // <-- Repetition of PORT was needed, this is a wireMock bug
```

Comment:
can you upgrade to `<version>1.58</version>` to see if the port is not repeated?
You can have as a tech-debt.

Reply:
#49 has been raised to deal with this.